src/main/java/com/springboot/star_wars_character_popularity/app/controller/CharacterController.java

```diff
@@ -13,9 +14,11 @@
 public class CharacterController {
 
     private CharacterService characterService;
+    private CharacterImportService characterImportService;
```

Comment:
Branimir told us to use final here.

Reply:
With that you can use @RequiredArgsConstructor from lombok, to remove boilerplate code.

Reply:
Final on both private CharacterService characterService;
 and private CharacterImportService characterImportService; ?

Reply:
@RequiredArgsConstructor
public class CharacterController {
private final CharacterService characterService;
private final CharacterImportService characterImportService;

this way, you don't need constructor, because of class annotation.

Reply:
Done.